src/mono/mono/mini/debugger-protocol.h

```diff
@@ -0,0 +1,367 @@
+	CMD_ASSEMBLY_GET_CATTRS = 14,
+	CMD_ASSEMBLY_GET_CUSTOM_ATTRIBUTES = 15
+} CmdAssembly;
+
+typedef enum {
+	CMD_MODULE_GET_INFO = 1,
+} CmdModule;
+
+typedef enum {
+	CMD_FIELD_GET_INFO = 1,
+} CmdField;
+
+typedef enum {
+	CMD_PROPERTY_GET_INFO = 1,
+} CmdProperty;
+
+typedef enum {
+	CMD_METHOD_GET_NAME = 1,
+	CMD_METHOD_GET_DECLARING_TYPE = 2,
+	CMD_METHOD_GET_DEBUG_INFO = 3,
+	CMD_METHOD_GET_PARAM_INFO = 4,
+	CMD_METHOD_GET_LOCALS_INFO = 5,
+	CMD_METHOD_GET_INFO = 6,
+	CMD_METHOD_GET_BODY = 7,
+	CMD_METHOD_RESOLVE_TOKEN = 8,
+	CMD_METHOD_GET_CATTRS = 9,
+	CMD_METHOD_MAKE_GENERIC_METHOD = 10,
+	CMD_METHOD_TOKEN = 11,
+	CMD_METHOD_ASSEMBLY = 12
+} CmdMethod;
+
+typedef enum {
+	CMD_TYPE_GET_INFO = 1,
+	CMD_TYPE_GET_METHODS = 2,
+	CMD_TYPE_GET_FIELDS = 3,
+	CMD_TYPE_GET_VALUES = 4,
+	CMD_TYPE_GET_OBJECT = 5,
+	CMD_TYPE_GET_SOURCE_FILES = 6,
+	CMD_TYPE_SET_VALUES = 7,
+	CMD_TYPE_IS_ASSIGNABLE_FROM = 8,
+	CMD_TYPE_GET_PROPERTIES = 9,
+	CMD_TYPE_GET_CATTRS = 10,
+	CMD_TYPE_GET_FIELD_CATTRS = 11,
+	CMD_TYPE_GET_PROPERTY_CATTRS = 12,
+	CMD_TYPE_GET_SOURCE_FILES_2 = 13,
+	CMD_TYPE_GET_VALUES_2 = 14,
+	CMD_TYPE_GET_METHODS_BY_NAME_FLAGS = 15,
+	CMD_TYPE_GET_INTERFACES = 16,
+	CMD_TYPE_GET_INTERFACE_MAP = 17,
+	CMD_TYPE_IS_INITIALIZED = 18,
+	CMD_TYPE_CREATE_INSTANCE = 19,
+	CMD_TYPE_GET_VALUE_SIZE = 20
+} CmdType;
+
+typedef enum {
+	CMD_STACK_FRAME_GET_VALUES = 1,
+	CMD_STACK_FRAME_GET_THIS = 2,
+	CMD_STACK_FRAME_SET_VALUES = 3,
+	CMD_STACK_FRAME_GET_DOMAIN = 4,
+	CMD_STACK_FRAME_SET_THIS = 5,
+	CMD_STACK_FRAME_GET_ARGUMENT = 6,
+} CmdStackFrame;
+
+typedef enum {
+	CMD_ARRAY_REF_GET_LENGTH = 1,
+	CMD_ARRAY_REF_GET_VALUES = 2,
+	CMD_ARRAY_REF_SET_VALUES = 3,
+	CMD_ARRAY_REF_GET_TYPE = 4
+} CmdArray;
+
+typedef enum {
+	CMD_STRING_REF_GET_VALUE = 1,
+	CMD_STRING_REF_GET_LENGTH = 2,
+	CMD_STRING_REF_GET_CHARS = 3
+} CmdString;
+
+typedef enum {
+	CMD_POINTER_GET_VALUE = 1
+} CmdPointer;
+
+typedef enum {
+	CMD_OBJECT_REF_GET_TYPE = 1,
+	CMD_OBJECT_REF_GET_VALUES = 2,
+	CMD_OBJECT_REF_IS_COLLECTED = 3,
+	CMD_OBJECT_REF_GET_ADDRESS = 4,
+	CMD_OBJECT_REF_GET_DOMAIN = 5,
+	CMD_OBJECT_REF_SET_VALUES = 6,
+	CMD_OBJECT_REF_GET_INFO = 7,
+	CMD_OBJECT_REF_GET_VALUES_ICORDBG = 8
+} CmdObject;
+
+typedef enum {
+	SUSPEND_POLICY_NONE = 0,
+	SUSPEND_POLICY_EVENT_THREAD = 1,
+	SUSPEND_POLICY_ALL = 2
+} SuspendPolicy;
+
+typedef enum {
+	CMD_EVENT_REQUEST_SET = 1,
+	CMD_EVENT_REQUEST_CLEAR = 2,
+	CMD_EVENT_REQUEST_CLEAR_ALL_BREAKPOINTS = 3
+} CmdEvent;
+
+typedef struct {
+	guint8 *buf, *p, *end;
+} Buffer;
+
+typedef struct {
+	int len;
+	int id;
+	int flags;
+	int command_set;
+	int command;
+	int error;
+	int error_2;
+} Header;
+
+typedef struct ReplyPacket {
+	int id;
+	int error;
+	Buffer *data;
+} ReplyPacket;
+
+/*
+FIXME:
+- Move EventKind back to debugger-agent.c as it contains sdb wire protocol constants.
+This is complicated because EventRequest has an event_kind field.
+
+*/
+
+typedef enum {
+	EVENT_KIND_VM_START = 0,
```

Comment:
Are these commands/events going to be used by the icordebug implementation as well ?

Reply:
Yes.